From a034ccaae4e76b3ca39beb31b6478d58115ce362 Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Mon, 31 Jan 2022 11:46:36 -0500
Subject: [PATCH 53/63] ima: Only emit error log messages for signed policy
 file for init_ima_ns

Only emite error log messages related to a signed policy file when
init_ima_ns is causing the error.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 security/integrity/ima/ima_fs.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/security/integrity/ima/ima_fs.c b/security/integrity/ima/ima_fs.c
index 58054e2dad14..feb0c71a4031 100644
--- a/security/integrity/ima/ima_fs.c
+++ b/security/integrity/ima/ima_fs.c
@@ -295,6 +295,7 @@ static const struct file_operations ima_ascii_measurements_ops = {
 
 static ssize_t ima_read_policy(struct user_namespace *user_ns, char *path)
 {
+	struct ima_namespace *ns = ima_ns_from_user_ns(user_ns);
 	void *data = NULL;
 	char *datap;
 	size_t size;
@@ -309,7 +310,8 @@ static ssize_t ima_read_policy(struct user_namespace *user_ns, char *path)
 	rc = kernel_read_file_from_path(path, 0, &data, INT_MAX, NULL,
 					READING_POLICY);
 	if (rc < 0) {
-		pr_err("Unable to open file: %s (%d)", path, rc);
+		if (ns == &init_ima_ns)
+			pr_err("Unable to open file: %s (%d)", path, rc);
 		return rc;
 	}
 	size = rc;
@@ -365,7 +367,8 @@ static ssize_t ima_write_policy(struct file *file, const char __user *buf,
 	if (data[0] == '/') {
 		result = ima_read_policy(user_ns, data);
 	} else if (ns->ima_appraise & IMA_APPRAISE_POLICY) {
-		pr_err("signed policy file (specified as an absolute pathname) required\n");
+		if (ns == &init_ima_ns)
+			pr_err("signed policy file (specified as an absolute pathname) required\n");
 		integrity_audit_msg(AUDIT_INTEGRITY_STATUS, NULL, NULL,
 				    "policy_update", "signed policy required",
 				    1, 0);
-- 
2.34.1

